src/FuncSharp/Try/Try.cs

```diff
@@ -49,83 +47,67 @@ public static Try<A> Create<A>(Func<Unit, A> f)
         /// <summary>
         /// Creates a new try with a successful result.
         /// </summary>
-        public static Try<A, E> Success<A, E>(A success)
+        public static TrySuccess<A> Success<A>(A value)
         {
-            return new Try<A, E>(success);
+            return new TrySuccess<A>(value);
         }
 
         /// <summary>
         /// Creates a new try with a successful result.
         /// </summary>
-        public static Try<A> Success<A>(A success)
+        public static Try<A, E> Success<A, E>(A value)
         {
-            return new Try<A>(success);
+            return new Try<A, E>(value);
         }
 
         /// <summary>
         /// Creates a new try with an error result.
         /// </summary>
-        public static Try<A, E> Error<A, E>(E error)
+        public static TryError<E> Error<E>(E error)
         {
-            return new Try<A, E>(error);
+            return new TryError<E>(error);
         }
 
         /// <summary>
-        /// Creates a new try with an exception result.
+        /// Creates a new try with an error result.
         /// </summary>
-        public static Try<A> Error<A>(Exception exception)
+        public static Try<A, E> Error<A, E>(E error)
```

Comment:
I think there might be problem with inference among these two methods, have you encountered any while writing it ?

Reply:
This version is for cases, where you explicitly want to specify full type without the implicit conversions. Might be useful in some more complicated places where the compiler cannot infer it easily.